awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,218 @@
+                          }
+    return counts
+
+
+@register('projects_by_scm_type')
+def projects_by_scm_type(since):
+    counts = dict(
+        (t[0] or 'manual', 0)
+        for t in models.Project.SCM_TYPE_CHOICES
+    )
+    for result in models.Project.objects.values('scm_type').annotate(
+        count=Count('scm_type')
+    ).order_by('scm_type'):
+        counts[result['scm_type'] or 'manual'] = result['count']
+    return counts
+
+@register('instance_info')   # maybe integrate this with job_instance_counts?
+def instance_info(since):
+    info = {}
+    instances = models.Instance.objects.values_list('hostname').annotate().values('uuid', 'version', 'capacity', 'cpu', 'memory', 'managed_by_policy', 'hostname')
+    for instance in instances:
+        info = {'uuid': instance['uuid'],
+                'version': instance['version'],
+                'capacity': instance['capacity'],
+                'cpu': instance['cpu'],
+                'memory': instance['memory'],
+                'managed_by_policy': instance['managed_by_policy'],
+                }
+    return info
+
+
+@register('job_counts')
+def job_counts(since):
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['status'] = dict(models.UnifiedJob.objects.values_list('status').annotate(Count('status')))
+    counts['launch_type'] = dict(models.UnifiedJob.objects.values_list('launch_type').annotate(Count('launch_type')))
+    
+    return counts
+    
+    
+@register('job_instance_counts')
+def job_instance_counts(since):         #TODO: all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+
+    job_types = models.UnifiedJob.objects.values_list(
+        'execution_node', 'launch_type').annotate(job_launch_type=Count('launch_type'))
+    for job in job_types:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+    job_statuses = models.UnifiedJob.objects.values_list(
+        'execution_node', 'status').annotate(job_status=Count('status'))
+    for job in job_statuses:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+# See if we can group by status and type inside the node grouping.  ^^
+
+    # 'jobs_running': models.UnifiedJob.objects.filter(execution_node=instance.hostname, status__in=('running', 'waiting',)).count(), # jobs in running & waiting state
+    # 'jobs_total': models.UnifiedJob.objects.filter(execution_node=instance.hostname).count(),
+    # ^^ try to get jobs_running and jobs_total per instance.  and uuid if easily possible  
+
+    return counts
+
+
+# Copies Job Events from db to a .csv to be shipped
+def copy_events_table(path='/tmp/'):
+    events_file = os.path.join(path, 'events_table.csv')
+    try:
+        write_data = open(events_file2, 'w')
+        with connection.cursor() as cursor:
+            cursor.copy_expert('''COPY (SELECT main_jobevent.id, 
+                                      main_jobevent.created, 
```

Comment:
@rooftopcellist there's a number of *keys* under `main_jobevent.event_data` that we *probably* want to capture here, but we _do not_ want the entire data structure.  What if we added some columns like this one:

```
SELECT main_jobevent.id,
    ...
    event_data::json->'task_action',
    event_data::json->'res'->'start',
    event_data::json->'res'->'end'
```

That said, @matburt @chrismeyersfsu this probably throws a *notable* performance wrench in because this column is `TEXT`, not `jsonb`, so we're going to have to parse these (which is going to potentially be slow/expensive past 24h).  That said, if we *really want* this data, we don't really have any other options (we _can't_ send the entire field value, because it potentially contains a _lot_ of sensitive data).

Reply:
@ryanpetrello What other keys from `main_jobevent.event_data` might we want?

Might we also want things from main_unifiedjob.job_args?

Reply:
No, `job_args` will very likely contain sensitive data if people aren't properly using `no_log`.  Let's collect the _bare minimum_ necessary to deliver the things @wenottingham has asked for.

Reply:
Specifically:

> What modules am I utilizing, and how much?  if 90% of your tasks are shell/command, it should gently chide you

...and:

> Are there anomalies in automation time for certain hosts, or certain playbooks?

@jladdjr I expect your work will be helpful here ^

> Are there anomalies in automation success for certain hosts or certain playbooks (flaky hosts/playbooks)

I think at a bare minimum her, we _probably_ want to capture *these* columns:

![image](https://user-images.githubusercontent.com/214912/54640712-d6a88a00-4a66-11e9-8032-f5067a3bbea3.png)


> How many people are using my corporate-approved roles, and how often?

`main_jobevent.role` should cover this for us, I think.

Reply:
> Are there anomalies in automation time for certain hosts, or certain playbooks?
>> @jladdjr I expect your perf plugin work will be most helpful here ^ Without that, we don't really have this type of data, historically.

@ryanpetrello - yes, definitely 👍 That sounds like it's right up the perf callback plugin's ally